pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -200,19 +203,71 @@ func (c *Reconciler) updateStatus(taskrun *v1alpha1.TaskRun) (*v1alpha1.TaskRun,
 	return newtaskrun, nil
 }
 
-// createBuild creates a build from the task, using the task's buildspec.
-func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun) (*buildv1alpha1.Build, error) {
+// createVolume will create a persistent volume mount for tr which
+// will be used to gather logs using the entrypoint wrapper
+func (c *Reconciler) createVolume(tr *v1alpha1.TaskRun) (*corev1.PersistentVolumeClaim, error) {
+	v, err := c.KubeClientSet.CoreV1().PersistentVolumeClaims(tr.Namespace).Create(
+		&corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: tr.Namespace,
+				// This pvc is specific to this TaskRun, so we'll use the same name
+				Name: tr.Name,
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: map[corev1.ResourceName]resource.Quantity{
+						corev1.ResourceStorage: *resource.NewQuantity(pvcSizeBytes, resource.BinarySI),
+					},
+				},
+			},
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create PVC %q: %s", tr.Name, err)
+	}
+	return v, nil
+}
+
+// createBuild creates a build from the task, using the task's buildspec
+// with volumeName as a volumeMount
+func (c *Reconciler) createBuild(tr *v1alpha1.TaskRun, volumeName string) (*buildv1alpha1.Build, error) {
 	// Get related task for taskrun
 	t, err := c.taskLister.Tasks(tr.Namespace).Get(tr.Spec.TaskRef.Name)
 	if err != nil {
 		return nil, fmt.Errorf("Error when listing tasks %v", err)
 	}
 
-	// TODO: Preferably use Validate on task.spec to catch validation error
-	if t.Spec.BuildSpec == nil {
-		return nil, fmt.Errorf("Task has nil BuildSpec")
+	// TODO: additional validation in Task webhook:
+	// No mounts with same names / path
+	// No env var with the name we'll be adding
```

Comment:
you can remove these comments too, i think we can handle this with randomization instead